tests/lib/__init__.py

```diff
@@ -262,6 +263,46 @@ def assert_installed(self, pkg_name, editable=True, with_files=[],
                 )
 
 
+def check_stderr(
+    stderr, expect_stderr_warning=False, expect_stderr_error=False,
+):
+    """
+    Check the given stderr for logged warnings and errors.
+
+    :param stderr: stderr output as a string.
+    :param expect_stderr_warning: whether a logged warning (or deprecation
+        message) is allowed.
+    :param expect_stderr_error: whether a logged error is allowed.  Passing
+        True for this argument implies `expect_stderr_warning` since warnings
+        are weaker than errors.
+    """
+    if expect_stderr_error:
+        expect_stderr_warning = True
```

Comment:
Passing explicitely `expect_stderr_warning=False, expect_stderr_error=True` should raise then (and `expect_stderr_warning` should default to `None`, meaning either `False` or `True` depending on `expect_stderr_error` value)